kmath-histograms/src/commonMain/kotlin/space/kscience/kmath/histogram/UniformHistogram1D.kt

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2018-2021 KMath contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package space.kscience.kmath.histogram
+
+import space.kscience.kmath.domains.DoubleDomain1D
+import space.kscience.kmath.misc.UnstableKMathAPI
+import space.kscience.kmath.operations.Group
+import space.kscience.kmath.operations.Ring
+import space.kscience.kmath.operations.ScaleOperations
+import space.kscience.kmath.operations.invoke
+import space.kscience.kmath.structures.Buffer
+import kotlin.math.floor
+
+@OptIn(UnstableKMathAPI::class)
+public class UniformHistogram1D<V : Any>(
+    public val group: UniformHistogram1DGroup<V, *>,
+    internal val values: Map<Int, V>,
+) : Histogram1D<Double, V> {
+
+    private val startPoint get() = group.startPoint
+    private val binSize get() = group.binSize
+
+    private fun produceBin(index: Int, value: V): Bin1D<Double, V> {
+        val domain = DoubleDomain1D((startPoint + index * binSize)..(startPoint + (index + 1) * binSize))
+        return Bin1D(domain, value)
+    }
+
+    override val bins: Iterable<Bin1D<Double, V>> get() = values.map { produceBin(it.key, it.value) }
+
+    override fun get(value: Double): Bin1D<Double, V>? {
+        val index: Int = group.getIndex(value)
+        val v = values[index]
+        return v?.let { produceBin(index, it) }
+    }
+}
+
+/**
+ * An algebra for uniform histograms in 1D real space
+ */
+public class UniformHistogram1DGroup<V : Any, A>(
+    public val valueAlgebra: A,
+    public val binSize: Double,
+    public val startPoint: Double = 0.0,
+) : Group<Histogram1D<Double, V>>, ScaleOperations<Histogram1D<Double, V>> where A : Ring<V>, A : ScaleOperations<V> {
+
+    override val zero: UniformHistogram1D<V> by lazy { UniformHistogram1D(this, emptyMap()) }
```

Comment:
`UniformHistogram1D` is very small, so it would be more effective to initialize it eagerly, because `lazy` actually creates a couple of objects too, hence should not be used unless underlying value is large or requires long calculations.